modules/developers-guide/pages/graphql-oriented-using.adoc

```diff
@@ -0,0 +1,1062 @@
+custom field types, such as a set of strings, `@cql_column(typeHint: "set<varchar>")`.
+// end::CreateObjectTypes[]
+
+// tag::CreateIndexes[]
+=== Creating indexes
+
+Cassandra supports indexing any regular, non-primary key fields in an object type.
+A field will  be indexed if `@cql_index` is added to the field definition.
+Indexed fields can be used as parameters in queries.
+
+[tabs]
+====
+
+Reader type::
++
+--
+[source,shell]
+----
+include::example$graphql/1createReaderIndex.graphql[]
+----
+--
+====
+
+The directive `@cql_index` has the following optional arguments:
+
+|====
+| Argument  | Default | Description
+| name | Generated | Custom index name
+| class | Secondary index | Custom index class, such as SAI
+| target | VALUES | Specifies set and list index type. Options are FULL and VALUES
+| options | N/A | Any options to pass to the underlying index query.
+|====
+// end::CreateIndexes[]
+
+// tag::CreateQueries[]
+=== Creating queries
+
+Most types in your schema will just be normal object types, but there are two types
+that are special, the Query type and the Mutation type.
+Every GraphQL service has a query type.
+Mutation types are optional.
+These types are special because they define the entry point of every GraphQL query.
+
+Queries
+
+[tabs]
+====
+Queries::
++
+--
+[source,shell]
+----
+include::example$graphql/1createQuery.graphql[]
+----
+--
+====
+
+These queries each specify the fields required and optional, as well as the input
+type.
+
+In the case of `book`, `title` is a required column, and `author` is an optional field
+input to retrieve a particular book. Because you specified `@cql_input` in the
+object type, a book can be the input to this query,
+while `[Book]` specifies that the output will be an array of books.
+// end::CreateQueries[]
+
+// tag::CreateMutations[]
+=== Creating mutations
+
+Mutations work in a similar way  to queries.
+Define fields in the mutation type, and you can specify those fields in your queries.
+
+[tabs]
+====
+Mutations::
++
+--
+[source,shell]
+----
+include::example$graphql/1createOperations.graphql[]
+----
+--
+====
+
+For mutations, notice the use of the input type `book:BookInput!` to define that the
+type submitted to the mutation is a `book`.
+`insertBook` will insert a book, `insertReader` will insert a reader, and `deleteBook`
+and `deleteReader` are Boolean operations.
+
+==== Conditional inserting
+
+A mutation can be turned into a conditional insertion operation by ending the mutation
+name with `ifNotExists`.
+However, it may be useful to know if the insertion is successfully applied.
+In that case, see payload type below to see how a boolean field `applied` can
+be added to the response from insertions.
+
+or annotating it with @cql_insert(ifNotExists: true)
+// end::CreateMutations[]
+
+// tag::CreatePayloadTypes[]
+=== Creating payload types
+
+Payloads types are not mapped to the database.
+They only serve as wrappers of an operation's arguments or response.
```

Comment:
Actually, only an operation's response at this point (my bad, I got it wrong in the draft I shared with you).